components/bsx/Create/CreateCollection.vue

```diff
@@ -62,143 +57,153 @@ import shouldUpdate from '@/utils/shouldUpdate'
-  BasicSwitch: () => import('@/components/shared/form/BasicSwitch.vue'),
-  SubmitButton: () => import('@/components/base/SubmitButton.vue'),
-  Money: () => import('@/components/bsx/format/TokenMoney.vue'),
-  AccountBalance: () => import('@/components/shared/AccountBalance.vue'),
-  MultiPaymentFeeButton: () =>
-    import('@/components/bsx/specific/MultiPaymentFeeButton.vue'),
-  NeoField,
+import Loader from '@/components/shared/Loader.vue'
+import BaseCollectionForm from '@/components/base/BaseCollectionForm.vue'
+import SubmitButton from '@/components/base/SubmitButton.vue'
+import Money from '@/components/bsx/format/TokenMoney.vue'
+import AccountBalance from '@/components/shared/AccountBalance.vue'
+import MultiPaymentFeeButton from '@/components/bsx/specific/MultiPaymentFeeButton.vue'
+
+const { $consola, $i18n } = useNuxtApp()
+const { balance } = useAuth()
+const { decimals, unit } = useChain()
+const { urlPrefix, tokenId } = usePrefix()
+const { apiUrl } = useApi()
+
+const { t } = $i18n
+
+const emit = defineEmits(['created'])
+
+const base: BaseCollectionType = {
+  name: '',
+  file: null,
+  description: '',
 }
+const id = '0'
+const attributes: Attribute[] = []
 
-@Component({ components })
-export default class CreateCollection extends mixins(
-  MetaTransactionMixin,
-  ChainMixin,
-  AuthMixin,
-  PrefixMixin,
-  ApiUrlMixin
-) {
-  public base: BaseCollectionType = {
-    name: '',
-    file: null,
-    description: '',
-  }
-  public collectionDeposit = ''
-  protected id = '0'
-  protected attributes: Attribute[] = []
-  protected balanceNotEnough = false
-  public feesToken: Token = 'BSX'
-  @Ref('collectionForm') readonly collectionForm
-
-  public checkValidity() {
-    return this.collectionForm.checkValidity()
-  }
+const collectionDeposit = ref('')
+const balanceNotEnough = ref(false)
+const feesToken = ref('BSX' as Token)
+const collectionForm = ref()
+const accountId = ref()
+const isLoading = ref(false)
+const status = ref('')
 
-  get balanceNotEnoughMessage() {
-    if (this.balanceNotEnough) {
-      return this.$t('tooltip.notEnoughBalance')
-    }
-    return ''
-  }
+const checkValidity = () => {
+  return collectionForm.value?.checkValidity()
+}
 
-  @Watch('accountId', { immediate: true })
-  async onAccountIdChange(val: string, oldVal: string) {
-    if (shouldUpdate(val, oldVal)) {
-      this.feesToken = await getFeesToken()
-    }
+const balanceNotEnoughMessage = computed(() => {
+  if (balanceNotEnough.value) {
+    return t('tooltip.notEnoughBalance')
   }
+  return ''
+})
 
-  public async created() {
-    onApiConnect(this.apiUrl, (api) => {
-      const classDeposit = getclassDeposit(api)
-      const metadataDeposit = getMetadataDeposit(api)
-      this.collectionDeposit = (classDeposit + metadataDeposit).toString()
-    })
-  }
-  get balanceOfToken() {
-    return getBalance(this.feesToken)
+watch(accountId, async (newVal, oldVal) => {
```

Comment:
can you check if this is being executed?
I don't see anywhere in the code that accountId is being changed.